### PR TITLE
Solar PV unit model

### PR DESCRIPTION
**Pull request by dguittet**
## Addresses issue:

Adds solar PV unit model, tests, and option in flowsheet model

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md and COPYRIGHT.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by bknueven — COMMENTED** [review]
Looks pretty straightforward. I just have a few minor comments.

File: dispatches/case_studies/renewables_case/load_parameters.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 # constants
 h2_mols_per_kg = 500
 H2_mass = 2.016 / 1000
+kg_to_tons = 0.00110231
```
[the remark this answers is not in the document]
Comment (dguittet) [reply dispatches/case_studies/renewables_case/load_parameters.py:27]: Not here but it does gets used in the flowsheet for arizona g&T, which is in a private repo

File: dispatches/case_studies/renewables_case/RE_flowsheet.py [hunk 2]
```diff
@@ -86,6 +87,22 @@ def add_wind(m, wind_mw, wind_resource_config=None):
...
+    Adds a solar pv unit to the flowsheet with a fixed system capacity and that uses resource data to determine the capacity factors
+
+    Args:
+        m: existing ConcreteModel with a flowsheet `fs`
+        pv_mw: capacity of PV model to be fixed
+        pv_resource_config: dictionary of Config keys `capacity_factors` and ConfigValues
+    Returns:
+        pv unit model in the flowsheet
+    """
+    m.fs.pv = SolarPV(**pv_resource_config)
```
[the remark this answers is not in the document]
Comment (dguittet) [reply dispatches/case_studies/renewables_case/RE_flowsheet.py:101]: thanks for catching that. I made `pv_resource_config` non-optional

File: dispatches/unit_models/solar_pv.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    except TypeError:
+        lst = [ListOf(float)(arg), ]
+    return lst
+
+
+def dict_of_list_of_list_of_floats(arg):
+    dic = dict()
+    for k, v in arg.items():
+        dic[k] = list_of_list_of_floats(v)
+    return dic
```
[the remark this answers is not in the document]
Comment (dguittet) [reply dispatches/unit_models/solar_pv.py:50]: Nope, removing

File: dispatches/case_studies/renewables_case/tests/test_RE_flowsheet.py [hunk 4]
```diff
@@ -99,7 +135,7 @@ def test_wind_battery_pem_optimize(input_params):
     assert design_res['pem_mw'] == pytest.approx(0, abs=1e-1)
     assert design_res['annual_rev_h2'] == pytest.approx(0.055, abs=1e-1)
     assert design_res['annual_rev_E'] == pytest.approx(531576401, rel=1e-2)
-    assert design_res['NPV'] == pytest.approx(2322131921, rel=1e-2)
+    assert design_res['NPV'] == pytest.approx(1009377998, rel=1e-2)
```
[the remark this answers is not in the document]
Comment (dguittet) [reply dispatches/case_studies/renewables_case/tests/test_RE_flowsheet.py]: Reverted this test change by undo-ing some changes made erroneously

**Review by bknueven — APPROVED** [review]

**Review by radhakrishnatg — APPROVED** [review]
Looks good to me.

**Pull request merged**